### ISSUE TITLE
Non deterministic training results

### ISSUE THREAD

**Issue by WangJuan6** [sunsmarterjie/yolov12#108]
Hi, I loaded the pre-trained yolov12n.pt model and fine-tuned it for 500 epochs on a single GPU using a custom dataset.
Despite setting seed=0 and deterministic=True, I still get different results after each training.
How can I get reproducible and deterministic results?
